packages/space-opera/src/components/mobile_view/components/mobile_expandable_section.ts

```diff
@@ -0,0 +1,157 @@
+  @query('me-checkbox#ar-modes') arModesCheckbox!: CheckboxElement;
+
+  renderDeployButton(): TemplateResult {
+    return html`
+    <mwc-button unelevated
+      icon="file_download"
+      ?disabled=${!this.isDeployable}
+      @click=${this.onInitialDeploy}>
+        Deploy Mobile
+    </mwc-button>`;
+  }
+
+  get optionalMessage(): TemplateResult {
+    const isOutOfSync = this.haveReceivedResponse &&
+        (!this.isSendingData && this.contentHasChanged);
+    if (isOutOfSync) {
+      return html`
+    <div style="color: #DC143C; margin-top: 5px;">
+      Your mobile view is out of sync with the editor.
+    </div>`;
+    } else if (this.isSendingData) {
+      return html`
+    <div style="color: white; margin-top: 5px;">
+      Sending data to mobile device. Textured models will take some time.
+    </div>`;
+    } else if (!this.haveReceivedResponse) {
+      return html`
+      <div style="color: white; margin-top: 5px;">
+        Use the QR Code to open the mobile viewing page on a mobile device.
+      </div>`;
+    }
+    return html``;
+  }
+
+  selectArMode() {
+    this.onSelectArMode!(this.arModesCheckbox.checked);
+  }
+
+  enableARChange() {
+    this.onEnableARChange!(this.arCheckbox.checked);
+  }
+
+  renderMobileInfo() {
+    const isOutOfSync = this.haveReceivedResponse &&
+        (!this.isSendingData && this.contentHasChanged);
+    const outOfSyncColor = isOutOfSync ? '#DC143C' : '#4285F4';
+    return html`
+    <div>
+      <mwc-button unelevated @click=${
+        this.openModal} style="margin-bottom: 10px;">
+        View QR Code
+      </mwc-button>
+      <mwc-button unelevated icon="cached" @click=${this.postInfo}
+        ?disabled=${
+    !this.haveReceivedResponse || this.isSendingData}
+        style="--mdc-theme-primary: ${outOfSyncColor}">
+        Refresh Mobile
+      </mwc-button>
+      ${this.optionalMessage}
+    </div>
+    <div style="font-size: 14px; font-weight: 500; margin: 16px 0px 10px
+    0px;">AR Settings:</div> <me-checkbox
+      id="ar-modes"
+      label="Default AR Modes to Scene Viewer"
+      ?checked="${this.defaultToSceneViewer}"
+      @change=${this.selectArMode}
+      >
+    </me-checkbox>
+    <me-checkbox
+      id="ar"
+      label="Enable AR"
+      ?checked="${!!this.arConfig!.ar}"
+      @change=${this.enableARChange}
+      >
+    </me-checkbox>
+    `;
+  }
+
+  renderIos() {
+    const needUsdzButton = this.iosAndNoUsdz ? '#DC143C' : '#4285F4';
+    return html`
+    <div style="font-size: 14px; font-weight: 500; margin: 16px 0px 10px
+    0px;">iOS Settings:</div> <mwc-button unelevated icon="file_upload"
+    @click=${this.onUploadUSDZ} style="--mdc-theme-primary:
+    ${needUsdzButton}">
+      USDZ
+    </mwc-button>
+    ${
+        this.iosAndNoUsdz ? html`
+  <div style="color: #DC143C; margin-top: 5px;">
+    Upload a .usdz to view model in AR on an iOS device.
+  </div>` :
+                            html``}
+    `
```

Comment:
I'm having a little trouble parsing all the ticks here. Can you move some of this out into their own `const` to make this easier to read? Also seem like the indentation is a bit weird.